content/5_Plat/Matrix_Expo.mdx

```diff
@@ -37,6 +37,90 @@ frequency: 1
 	</Resource>
 </Resources>
 
+
+## Example - Fibonacci
+
+<FocusProblem problem='sample' />
+
+The fibonacci numbers are defined by the following matrix relation
+
+$$
+A = \begin{bmatrix}1&1\\1&0\end{bmatrix}^n=\begin{bmatrix}F_{n+1} & F_n \\ F_n & F_{n-1}\end{bmatrix}
+$$
+
+### Proof by Induction
+
+The fibonacci numbers are defined as follows:
+
+$$
+\begin{align*}
+F_0 &= 0 \\
+F_1 &= 1 \\
+F_{n+1} &= F_{n} + F_{n-1}
+\end{align*}
+$$
+
+Base case ($n=1$):
+
+$$
+A^1 = \begin{bmatrix}1 & 1 \\ 1 & 0\end{bmatrix} = \begin{bmatrix}F_2 & F_1 \\ F_1 & F_0\end{bmatrix}
+$$
+
+Induction step ($n+1$):
+
+$$
+A^{n+1}=A\,A^n=\begin{bmatrix}
+1 & 1 \\ 1 & 0
+\end{bmatrix}\begin{bmatrix}
+F_{n+1} & F_{n} \\ F_{n} & F_{n-1}
+\end{bmatrix}=\begin{bmatrix}
+F_{n+1}+F_n & F_{n}+F_{n-1} \\  F_{n+1} & F_{n}
+\end{bmatrix}=\begin{bmatrix}
+F_{n+2} & F_{n+1} \\ F_{n+1} & F_{n}
+\end{bmatrix}
+$$
+
+The base case is true, and the induction step is true. Therefore the formula
+holds for all $n \in \mathbb{N}$
+
+### Implementation
+
+**Time complexity:** $\mathcal{O}(\log N)$
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+using ll = long long;
+
+const ll MOD = (ll)1e9 + 7;
+
+vector<ll> mul(vector<ll> a, vector<ll> b) {
```

Comment:
why not `array<array<ll,2>,2>`?

Reply:
it was easier to type but I'll go change it real quick